<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MazeGui</class>
 <widget class="QMainWindow" name="MazeGui">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1066</width>
    <height>692</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Maze Solver</string>
  </property>
  <property name="windowIcon">
   <iconset theme="MazeSolver">
    <normaloff>.</normaloff>.</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="autoFillBackground">
    <bool>false</bool>
   </property>
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>30</y>
      <width>401</width>
      <height>196</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="userinputGridlayout">
     <item row="2" column="0">
      <widget class="QComboBox" name="algorithmCombobox">
       <item>
        <property name="text">
         <string>Wall Follower</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Tremaux's algorithm</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Recursive</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Shortest path algorithm</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="TimeLeft">
       <property name="frameShape">
        <enum>QFrame::Box</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Sunken</enum>
       </property>
       <property name="text">
        <string>Time left</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QLCDNumber" name="timeLeftLCDDisplay"/>
     </item>
     <item row="0" column="0" colspan="2">
      <widget class="QLabel" name="chooseAlgorithm">
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Plain</enum>
       </property>
       <property name="text">
        <string>Choose Algorithm</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QLCDNumber" name="stepLCDDisplay"/>
     </item>
     <item row="2" column="1">
      <widget class="QPushButton" name="simulationStartButton">
       <property name="text">
        <string>Start Simulation</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QPushButton" name="generateMazeButton">
       <property name="text">
        <string>Generate Maze</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <layout class="QHBoxLayout" name="HeightSpinbox">
       <item>
        <widget class="QLabel" name="widthLabel">
         <property name="text">
          <string>Width</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="widthSpinBox">
         <property name="maximum">
          <number>100000</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="heightLabel">
         <property name="text">
          <string>Height</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="heightSpinBox">
         <property name="maximum">
          <number>100000</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="steps">
       <property name="frameShape">
        <enum>QFrame::Box</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Sunken</enum>
       </property>
       <property name="text">
        <string>Steps</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QPushButton" name="resetButton">
       <property name="text">
        <string>Reset</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QFrame" name="frame">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>20</y>
      <width>641</width>
      <height>591</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>641</width>
      <height>0</height>
     </size>
    </property>
    <property name="sizeIncrement">
     <size>
      <width>1</width>
      <height>1</height>
     </size>
    </property>
    <property name="frameShape">
     <enum>QFrame::StyledPanel</enum>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <widget class="QGraphicsView" name="mazeGraphicsView">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>661</width>
       <height>571</height>
      </rect>
     </property>
     <property name="minimumSize">
      <size>
       <width>661</width>
       <height>0</height>
      </size>
     </property>
     <property name="autoFillBackground">
      <bool>true</bool>
     </property>
     <property name="verticalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOn</enum>
     </property>
     <property name="horizontalScrollBarPolicy">
      <enum>Qt::ScrollBarAlwaysOn</enum>
     </property>
     <property name="sizeAdjustPolicy">
      <enum>QAbstractScrollArea::AdjustToContentsOnFirstShow</enum>
     </property>
     <property name="backgroundBrush">
      <brush brushstyle="Dense7Pattern">
       <color alpha="255">
        <red>0</red>
        <green>0</green>
        <blue>0</blue>
       </color>
      </brush>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1066</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuLabyrinth_Algorithm">
    <property name="title">
     <string>Menu</string>
    </property>
   </widget>
   <addaction name="menuLabyrinth_Algorithm"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionGenerateMaze">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>SetMaze</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
